pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/MLTransactionMetadataStore.java

```diff
@@ -374,8 +374,8 @@ public CompletableFuture<Void> updateTxnStatus(TxnID txnID, TxnStatus newStatus,
                                 if (newStatus == TxnStatus.COMMITTED || newStatus == TxnStatus.ABORTED) {
                                     transactionLog.deletePosition(txnMetaListPair.getRight()).whenComplete((v, ex) -> {
                                         if (ex != null) {
-                                            promise.completeExceptionally(ex);
-                                            return;
+                                            log.warn("Failed to delete transaction log position "
+                                                    + "at end transaction [{}]", txnID);
```

Comment:
If we are not able to ack the message at this time, do we have other places to make sure the data can be removed?

Reply:
When broker restart, in `MLTransactionMetadataStore` init method,  cursor will be checked again. But if cursor does not work for a long time, it will cause broker restart difficulties.

https://github.com/apache/pulsar/blob/ac6bd3c71e24b90826438cd44395ca21a849067f/pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/MLTransactionMetadataStore.java#L161-L176

Reply:
Maybe we can try to restart `cursor` multiple times and stop providing transaction support when it finally fails.

Reply:
Maybe we should change the way we clean data. Change the way from individual Ack to cumulative Ack.

Reply:
Confirmed with @congbobo184 , there is no problem here, it will only fail when persisting markDelete, not when operating memory.

Reply:
@liangyepianzhou Ok, I see.